jdaviz/core/template_mixin.py

```diff
@@ -1490,12 +1491,14 @@ def _update_subset(self, subset, attribute=None):
                               else self._subset_to_dict(subset)
                               for s in self.items]
 
-        if attribute == 'subset_state' and subset.label == self.selected:
+        if attribute == 'subset_state' and subset.label in self.selected:
             # updated the currently selected subset
             self._clear_cache("selected_obj", "selected_item")
             self._update_has_subregions()
 
     def _update_has_subregions(self):
+        if self.is_multiselect:
+            self.selected_has_subregions = False
```

Comment:
I think the block after would fail in multiselect, right?

```suggestion
            self.selected_has_subregions = False
            return
```